motto/llm_gate.py

```diff
@@ -83,18 +83,22 @@ def get_func_def(fn, metta, prompt_space):
         }
     }
 
+
 def get_llm_args(metta: MeTTa, prompt_space: SpaceRef, *args):
     agent = None
     messages = []
     functions = []
     msg_atoms = []
-    def __msg_update(ag, m, f, a):
-        nonlocal agent, messages, functions, msg_atoms
+    docs_count = None
+    def __msg_update(ag, m, f, a, d):
+        nonlocal agent, messages, functions, msg_atoms, docs_count
```

Comment:
Why is it necessary to introduce `docs_count`? It looks very specific to the retrieval agent. Why should it be in `llm_gate`?

Reply:
> Why is it necessary to introduce `docs_count`? It looks very specific to the retrieval agent. Why should it be in `llm_gate`? I may guess that this is a sort of iterator over documents. But why can't it be done internally in the retrieval agent itself?
I can move docs_count to this line https://github.com/besSveta/metta-motto/blob/b44ccfcb820e5c1fe9a1a3ca27db011190e72d39/examples/test_retrieval.metta#L2C7-L2C7 
Or you think that the number of retrieved documents should be constant ?

Reply:
OK, I see. So, the call to the agent may need some parameterization. I'd consider a general approach for passing parameters without hardcoding each particular parameter into llm-gate. For now, the number of documents is better to move to the agent initialization from `llm` call. When the general mechanism for passing parameters is implemented, docs_count can be moved there.